### PR TITLE
Use normal for loop instead of map+fold for populating PCRs

### PR DESCRIPTION
**Pull request by jhand2**
Depending on optimizations applied by the compiler, it sometimes inserts a panic in the PCR accumulation code that depends on map/fold.

It is unclear in which cases it will insert one and which it won't, but for simplicity, replace this code with a for loop.

### CODE REVIEW

File: runtime/src/pcr.rs [hunk 1]
```diff
@@ -50,24 +50,18 @@ impl GetPcrQuoteCmd {
...
+        let mut pcrs = [[0u8; 48]; 32];
+        for (i, p) in raw_pcrs.iter().enumerate() {
+            pcrs[i] = p.into()
+        }
 
         Ok(MailboxResp::QuotePcrs(QuotePcrsResp {
             hdr: MailboxRespHeader::default(),
             nonce: args.nonce,
-            pcrs: pcrs_as_bytes,
+            pcrs: pcrs,
```
Comment (sree-revoori1) [inline runtime/src/pcr.rs:64]: nit: No need to do `pcrs: pcrs`, can just do `pcrs,`

File: runtime/src/pcr.rs [hunk 2]
```diff
@@ -15,7 +15,7 @@ Abstract:
 use crate::Drivers;
 use caliptra_cfi_derive_git::cfi_impl_fn;
 use caliptra_common::mailbox_api::{
-    ExtendPcrReq, IncrementPcrResetCounterReq, MailboxResp, MailboxRespHeader, QuotePcrsReq,
+    ExtendPcrReq, IncrementPcrResetCounterReq, MailboxResp, MailboxRespHeader, PcrValue, QuotePcrsReq,
```
Comment (sree-revoori1) [inline runtime/src/pcr.rs:18]: Is this import necessary?

Comment (jhand2) [reply runtime/src/pcr.rs]: No, removed

File: runtime/src/pcr.rs (hunk 1 above)
Comment (jhand2) [reply runtime/src/pcr.rs]: Done

**Review by sree-revoori1 — APPROVED** [review]

**Review by rusty1968 — APPROVED** [review]

**Pull request merged**